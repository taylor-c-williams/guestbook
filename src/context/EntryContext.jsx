import { createContext, useState } from 'react';

const EntryContext = createContext();
const EntryProvider = ({ children }) => {
  const [entries, setEntries] = useState([]);
  return (
    <EntryContext.Provider value={{ entries, setEntries }}>
      {children}
    </EntryContext.Provider>
  );
};
export { EntryProvider, EntryContext };
