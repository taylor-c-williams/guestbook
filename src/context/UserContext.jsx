import { createContext, useState } from 'react';

const UserContext = createContext();
const UserProvider = ({ children }) => {
  const [user, setUser] = useState();

  const login = (username, password) => {
    const loginSuccessful =
      username === process.env.AUTH_USERNAME &&
      password === process.env.AUTH_PASSWORD;
    if (loginSuccessful) setUser({ username });
    return loginSuccessful;
  };

  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
};
export { UserProvider, UserContext };
