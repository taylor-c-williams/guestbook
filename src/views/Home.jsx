import React from 'react';
import Header from '../components/Header/Header';
import EntryForm from '../components/EntryForm/EntryForm';
import EntryList from '../components/EntryList/EntryList';
import { EntryProvider } from '../context/EntryContext';

export default function Home() {
  return (
    <EntryProvider>
      {/* <Header /> */}
      sup
      <EntryForm />
      <EntryList />
    </EntryProvider>
  );
}
