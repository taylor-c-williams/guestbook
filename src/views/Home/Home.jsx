import React from 'react';
import Header from '../../components/Header/Header';
import Guestbook from '../../views/Guestbook/Guestbook';
import { EntryProvider } from '../../context/EntryContext';

export default function Home() {
  return (
    <main>
      <Header />
      <Guestbook />
    </main>
  );
}
