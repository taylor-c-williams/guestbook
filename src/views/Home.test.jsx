import React from 'react';
import { render } from '@testing-library/react';
import Home from './Home';
import { UserContext } from '../context/UserContext';

it('should render the Home view', () => {
  const { container } = (
    <UserContext>
      <Home />
    </UserContext>
  );
  expect(container).toMatchSnapshot();
});
