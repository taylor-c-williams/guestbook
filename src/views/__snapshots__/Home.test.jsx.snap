// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Home view 1`] = `
<div>
  sup
  <div />
  <div />
</div>
`;
